File: solutions/0100_Same_Tree/iter_bfs.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public boolean isSameTree(TreeNode p, TreeNode q) {
        Deque<TreeNode> s1 = new ArrayDeque<>();
        Deque<TreeNode> s2 = new ArrayDeque<>();
        if (!addBoth(s1, s2, p, q)) {
            return false;
        }
        while (s1.size() > 0 && s2.size() > 0) {
            TreeNode t1 = s1.pop();
            TreeNode t2 = s2.pop();
            if (!addBoth(s1, s2, t1.left, t2.left)) {
                return false;
            }
            if (!addBoth(s1, s2, t1.right, t2.right)) {
                return false;
            }
        }
        return s1.size() == s2.size();
    }

    private boolean addBoth(Deque<TreeNode> s1, Deque<TreeNode> s2,
                        TreeNode t1, TreeNode t2) {
        if (t1 == null && t2 == null) {
            return true;
        }
        if (t1 != null && t2 != null && t1.val == t2.val) {
            s1.push(t1);
            s2.push(t2);
            return true;
        }
        return false;
    }
}

File: solutions/0102_Binary_Tree_Zigzag_Level_Order_Traversal/iter_bfs.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

from functools import reduce
class Solution:
    def zigzagLevelOrder(self, root: TreeNode) -> List[List[int]]:
        if not root:
            return []
        ret, l = [], []
        l += [root]
        reverse_order = 0
        while l:
            new_ret, new_l = [], []
            for node in l:
                new_ret += [node.val]
                if node.left:
                    new_l += [node.left]
                if node.right:
                    new_l += [node.right]
            l = new_l
            ret += [list(reversed(new_ret))] if reverse_order else [new_ret]
            reverse_order = 1 - reverse_order
        return ret
